### DISCUSSION TITLE
podcast:guid | podcastnamespace.org

### DISCUSSION THREAD

**Discussion by giscus** [jamescridland/podcastnamespace.org#5] [General]
# podcast:guid | podcastnamespace.org

Adding new features to podcasting since 2020

https://podcastnamespace.org/tag/guid

**Comment by boryskozielski** [comment]
I'm using Wikidata to set guid. Ex: https://www.wikidata.org/wiki/Q113688101 Do you think it make sense?

**Reply by jamescridland** [reply]
No - the GUID is programmatically set from the (initial) RSS feed. [The specification](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#guid) has the details of how to make a GUID from an RSS feed. If the RSS feed changes in future, the GUID never changes.

**Comment by rastacalavera** [comment]
I am trying to learn more about GUIDs and their importance/value. I understand each show in the podcast index is given one, but there can also be GUIDs per episode. I haven't seen documentation around episode based GUIDs and as of now my rss generator just uses the my TLD + where my mp3 resides (selfhosting). Should I be generating GUIDs for each episode I release and if so, what is the proper method to do it?

**Reply by jamescridland** [reply]
For an episode GUID, that can be anything as long as it is different for every episode, and remains constant. But you need an episode GUID.

Armed with a podcast GUID and an episode GUID, you should always be able to find that one episode, whatever RSS feed you end up being on.
